app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.from_string(mode_string)
```

Comment:
Another option would be `def self.[](mode_string)`, which would use the semantics of `UserNoteMode["required"]`, which is fairly idiomatic, though rare. There's also the ["conversion function"](http://www.virtuouscode.com/2012/05/07/a-ruby-conversion-idiom/) `UserNoteMode("required")`, but I find the setup for that annoying, and I would probably only want to use that if we were supporting converting from multiple formats (e.g. string and integer).

I may still change the name of this method to be able to take a `UserNoteMode` instance to make it idempotent, but I may wait on that until I write some specs.

```
if mode.is_a?(UserNoteMode)
  mode
elsif VALID_MODES.include?(mode)
  new(mode)
else
  InvalidUserNoteMode.new(mode_string)
end
```